code/visualize_spline_fits.R

```diff
@@ -0,0 +1,153 @@
+WHERE country = 'Portugal'
+AND lineage in ('BA.2', 'BA.4', 'BA.5', 'BA.2.12.1', 'BA.1', 'other')
+")) %>%
+    expand(collection_date, lineage, reference_date)
+
+# Make available for  joining
+dbWriteTable(con, "grid", grid)
+
+d <- dbGetQuery(con,
+                 paste0("
+SELECT
+pred.lineage,
+pred.reference_date,
+pred.country,
+pred.collection_date,
+pred.p_hat,
+dat.p_lineage,
+dat.p_lineage_se
+FROM grid g
+FULL JOIN
+(SELECT
+        lineage,
+        country,
+        collection_date,
+        p_lineage,
+        p_lineage_se
+    FROM '", baseline, "'
+    WHERE lineage IN ('BA.2', 'BA.4', 'BA.5', 'BA.2.12.1', 'BA.1', 'other')) dat
+ON g.lineage = dat.lineage
+AND g.collection_date = dat.collection_date
+FULL JOIN (SELECT *,
+      regexp_extract(filename, '[0-9]*-[0-9]*-[0-9]*', 0) :: DATE AS reference_date
+      FROM read_parquet('", predicted, "', filename = true)) pred
+ON pred.lineage = dat.lineage
+AND pred.country = dat.country
+AND pred.collection_date = dat.collection_date
+AND pred.reference_date = g.reference_date
+WHERE pred.country in ('Portugal', 'Brazil', 'Spain', 'India', 'Czechia')
+AND pred.lineage IN ('BA.2', 'BA.4', 'BA.5', 'BA.2.12.1', 'BA.1', 'other')
+                        ")) 
+
+
+for (lineage_to_show in c('BA.2', 'BA.4', 'BA.5', 'BA.2.12.1', 'BA.1', 'other')) {
+ 
+    
+    p1 <- d %>%
+    filter(lineage == 'BA.5') %>%
+    ggplot() +
+    geom_point(aes(collection_date,
+                   as.double(p_lineage),
+                   alpha = if_else(collection_date < reference_date, 0.25, 1)),
+               show.legend = FALSE) +
+    geom_line(aes(collection_date, p_hat, color = lineage)) +
+    geom_vline(aes(xintercept = reference_date),
+               alpha = 0.2) +
+    facet_grid(country~reference_date) +
+    theme_bw()
+
+    ggsave(paste0('figs/supp/spline_supp_', lineage_to_show, '.png'), p1)
+
+}
+
+p1 <- d %>%
+    filter(country == 'India') %>%
+    ggplot() +
+    geom_point(aes(collection_date,
+                   as.double(p_lineage),
+                   alpha = if_else(collection_date < reference_date, 0.25, 1)),
+               show.legend = FALSE) +
+    geom_line(aes(collection_date, p_hat, color = lineage)) +
+    geom_vline(aes(xintercept = reference_date),
+               alpha = 0.2) +
+    facet_grid(lineage~reference_date) +
+    theme_bw()
+
+    ggsave(paste0('figs/supp/spline_supp_', 'india', '.png'), p1)
+
+
+
+p2 <- dbGetQuery(con,
+           paste0("
+SELECT *,
+regexp_extract(filename, '[0-9]*-[0-9]*-[0-9]*', 0) :: DATE AS reference_date
+FROM read_parquet('", contrast, "', filename  = TRUE)
+                  ")) %>%
+    ggplot(aes(reference_date, estimate))+
+    geom_line()+
+    geom_ribbon(aes(x = reference_date,
+                ymin = estimate - 2 * SE,
+                ymax = estimate + 2 * SE),
+                alpha = 0.2) +
+    facet_wrap(~contrast)+
+    theme_bw()
+
+ggsave('figs/supp/contrasts.png', p2)
```

Comment:
Label the y-axis!

Reply:
Maybe add a title so we know this is BA.5 to BA.2. And this is the median of the multicountry model you're showing?